File: Python/ano_novo.py
from datetime import datetime, timedelta

data_hora = input() #2023-01-01 19:00:00
fuso_horario_local = int(input()) #-3 
cidade_destino = input() #Grécia, Atenas 
fuso_horario_destino = int(input()) #2 

data_hora_local = datetime.strptime(data_hora, '%Y-%m-%d %H:%M:%S')
data_hora_local += timedelta(hours=fuso_horario_local + fuso_horario_destino - 3)
data_hora_destino = datetime.strptime(data_hora, '%Y-%m-%d %H:%M:%S') + timedelta(hours=fuso_horario_destino)
data_hora_destino = data_hora_destino.replace(hour=0, minute=0, second=0, microsecond=0)

print("Já é Ano Novo em " + cidade_destino + "? " + str(data_hora_destino.year == data_hora_local.year).lower)

File: JS/anoNovo.js
import prompt from 'readline';
import util from 'util';
const rl = prompt.createInterface({ input: process.stdin, output: process.stdout });
const question = util.promisify(rl.question).bind(rl);
const read = (msg) => question(`${msg?msg+'\n':''}`);




let dataHora = await read(); //2023-01-01 19:00:00
let fusoHorarioLocal =  Number(await read()); //-3 
let cidadeDestino = await read(); //Grécia, Atenas 
let fusoHorarioDestino = Number(await read()); //2 

let dataHoraLocal = new Date(dataHora);
dataHoraLocal.setHours(Number(dataHoraLocal.getHours()) + fusoHorarioLocal + fusoHorarioDestino - 3);
let dataHoraDestino = new Date(new Date(dataHora).toLocaleString('en-US', {timeZone: 'Europe/Athens'}));
console.log("Já é Ano Novo em " + cidadeDestino + "? " + (dataHoraDestino.getFullYear() == dataHoraLocal.getFullYear()));
